Sources/NIO/EventLoopFuture.swift

```diff
@@ -1325,3 +1325,77 @@ extension EventLoopFuture {
         return self
     }
 }
+
+// MARK: unwrap
+
+extension EventLoopFuture {
+    /// Unwrap an `EventLoopFuture` where its type parameter is an `Optional`.
+    ///
+    /// Unwraps a future returning a new `EventLoopFuture` with the same value as the resolved future when its value
+    /// is `Optional.some(...)`. otherwise the `Error` passed in the `or` parameter is thrown. For example:
+    /// ```
+    /// do {
+    ///     try promise.futureResult.unwrap(or: ErrorToThrow).wait()
+    /// } catch ErrorToThrow {
+    ///     ...
+    /// }
+    /// ```
+    ///
+    /// - parameters:
+    ///     - or: the `Error` that is thrown when then resolved future's value is `Optional.none`.
+    /// - returns: an new `EventLoopFuture` with new type parameter `T` and the same value as the resolved
+    ///     future.
+    /// - throws: the `Error` passed in the `or` parameter when the resolved future's value is `Optional.none`.
+    func unwrap<T>(or error: Error) -> EventLoopFuture<T> where Value == T? {
```

Comment:
Two minor nits: I'd rather spell this `T?` as `Optional<T>`, and then can we use `NewValue` as the generic type parameter? Single letter type parameters aren't great.